packages/zowe-explorer-api/src/vscode/ui/utils/TableBuilder.ts

```diff
@@ -0,0 +1,224 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { ExtensionContext } from "vscode";
+import { Table } from "../TableView";
+import { TableMediator } from "./TableMediator";
+
+/**
+ * A builder class for quickly instantiating {@link Table.View} instances.
+ * Especially useful for building multiple tables with similar layouts or data.
+ *
+ * @remarks
+ *
+ * ## Building a table
+ *
+ * Developers can build a table using the helper methods provided by the builder.
+ * Once a table is ready to be built, use the {@link TableBuilder.build} function to create
+ * a new {@link Table.View} instance with the given configuration.
+ *
+ * ## Sharing tables
+ *
+ * Share a table during the build process by using the {@link TableBuilder.buildAndShare} function.
+ * This function will create the {@link Table.View} instance while also adding it to the mediator
+ * in Zowe Explorer's extender API. Extenders who would like to contribute to shared tables can do
+ * so by accessing the table by its unique ID.
+ */
+export class TableBuilder {
+    private context: ExtensionContext;
+    private data: Table.ViewOpts = {
+        actions: {
+            all: [],
+        },
+        contextOpts: {
+            all: [],
+        },
+        columns: [],
+        rows: [],
+        title: "",
+    };
+    private forWebviewView = false;
+
+    public constructor(context: ExtensionContext) {
+        this.context = context;
+    }
+
+    public isView(): this {
+        this.forWebviewView = true;
+        return this;
+    }
+
+    /**
+     * Set optional properties for the table view.
+     * @param opts The options for the table
+     * @returns The same {@link TableBuilder} instance with the options added
+     */
+    public options(opts: Table.GridProperties): this {
+        this.data = { ...this.data, options: this.data.options ? { ...this.data.options, ...opts } : opts };
+        return this;
+    }
+
+    /**
+     * Set the title for the next table.
+     * @param name The name of the table
+     * @returns The same {@link TableBuilder} instance with the title added
+     */
+    public title(name: string): this {
```

Comment:
Would it make sense to make this a setter? This also applies to a few other properties here.

Reply:
Currently all of the `TableBuilder` functions are meant to return the same instance, that way you can chain several methods to get the final table object:

```typescript
const table = new TableBuilder()
    .title("A table")
    .rows([{ a: 1, b: 2, c: 3 ])
    .columns([{ field: "a" }, { field: "b" }, { field: "c" }])
    .options({ pagination: false, paginationPageSize: 0 })
    .build();
```
By changing it to a setter, you would have to save the builder to an object, insert a new line to assign the title, and then continue re-assigning that object.

That being said, if we feel that its more appropriate to use setters for functions for specific properties (such as title), I wouldn't be opposed to incorporating those changes 😋

Reply:
Let's just keep it as-is. I think this is just a nuanced difference in the code style/structure.
I do like the concatenation as it reminds me of other personal, more structurally relaxed projects 😋 

---

> By changing it to a setter, you would have to save the builder to a let-scoped variable, insert a new line to assign the title, and then continue to re-assign that variable

The above is only true if we has a combination of these setters and concatenation-friendly methods. I understand the convenience of returning `this` everywhere, it's just a slightly different paradigm of how the rest of the codebase might be structured.

Technically having them all as setters may look something like this:

```typescript
const builder = new TableBuilder(); 
builder.title = "A table"; 
builder.rows = [ { a: 1, b: 2, c: 3} ] 
builder.columns = [{ field: "a" }, { field: "b" }, { field: "c" }]
builder.options = { pagination: false, paginationPageSize: 0 };
const table = builder.build();
```

And a slightly different approach might be to allow for the table properties to be provided at the constructor level instead of having to concatenate calls, or split statements.

```typescript
const table = new TableBuilder({ 
  title: "A table",
  rows: [ { a: 1, b: 2, c: 3} ],
  columns: [{ field: "a" }, { field: "b" }, { field: "c" }],
  options: { pagination: false, paginationPageSize: 0 },
}).build();
```

Reply:
My approach here was definitely a bit biased because it feels similar to coding in Rust, where libraries often use this pattern to allow chaining functions on one line 😆 That said, the other 2 options are definitely viable.

Something to keep in mind for the "setter" approach is that we'd also need to implement getters for each "property" to ensure some capabilities are still supported (e.g. adding new rows vs. replacing all rows). Definitely not a dealbreaker, just something to consider.

Also, the constructor approach definitely feels more TypeScript-like, but prevents making changes without constructing a new object.

Reply:
I struggled a bit in my head on this one.
But I think this is what swayed me into keeping it as-is (with the method-chaining approach) was this comment right here...

> the constructor approach feels more akin to a typical TypeScript utility, but it prevents making changes without constructing a new builder object.

Thanks for the discussion Trae 🙏🏽 
And sorry for being difficult 😅